extension/services/spotify/player.js

```diff
@@ -11,8 +11,23 @@ this.player = (function() {
     }
 
     async action_search({ query, thenPlay }) {
-      const searchButton = this.querySelector("a[aria-label='Search']");
-      searchButton.click();
+      try {
+        const searchButton = this.querySelector("a[aria-label='Search']");
+        searchButton.click();
+      } catch (error) {
+        this.log(error);
+      }
+      try {
+        this.querySelector("div.ErrorPage");
+        return new Promise((resolve, reject) => {
+          const e = new Error(`You must enable DRM.`);
+          e.displayMessage = `You must enable DRM.`;
+          e.name = "DRMNotEnabled";
+          reject(e);
+        });
+      } catch (error) {
+        this.log(error);
+      }
```

Comment:
So to clarify, I would first try to select the `div.ErrorPage`, if it exists, throw a new `Error` with `.displayMessage = You must enable DRM`. 

- what (if any) errors should be logged here?

Otherwise, the process of selecting the search button would continue.

Reply:
I realized `.displayMessage` actually doesn't work here. I added a note here about how it should work:

https://github.com/mozilla/firefox-voice/pull/1178#discussion_r389383014

In general right here you can simply throw that `new Error(...)` and you don't have to catch it here. Instead you should catch it in `spotify.js` and then check if `e.message === "You must enable DRM."` – and if it does, add that .displayMessage there.

Reply:
Looking at [#1178](https://github.com/mozilla/firefox-voice/pull/1178#discussion_r389383014), would something like this be what you're looking for?

`spotify.js`
```js
    try {
      await this.initTab("/services/spotify/player.js");
    } catch (err) {
      console.debug(err);
      if (err.message === 'You must enable DRM.') {
        err.displayMessage = 'You must enable DRM.';
      }
    }
```

`player.js`
```js
      try {
        const searchButton = this.querySelector("a[aria-label='Search']");
        searchButton.click();
      } catch (err) {
        if (err.name === 'ElementNotFound') {
          const err = new Error(`You must enable DRM.`);
          throw err;
        }
      }
```

Reply:
Small suggestion: the err should be thrown in the catch block regardless of `err.name`.

Reply:
There's probably other reasons like a login page or network error or all sorts of things that would keep the Search button from being available, so it would be better to actually look for the DRM error message specifically.

Your code in `spotify.js` looks right, yes

Reply:
I've made the necessary changes now, hopefully things look good.